include/material/mohr_coulomb.tcc

```diff
@@ -511,33 +506,33 @@ Eigen::Matrix<double, 6, 1> mpm::MohrCoulomb<Tdim>::compute_stress(
   double yield = 0.;
   if (yield_type_trial == FailureState::Tensile)
     yield = yield_function_trial(0);
-  if (yield_type_trial == FailureState::Shear)
-    yield = yield_function_trial(1);
+  if (yield_type_trial == FailureState::Shear) yield = yield_function_trial(1);
+
   double lambda_trial =
       yield /
       ((df_dsigma_trial.transpose() * de_).dot(dp_dsigma_trial.transpose()) +
        softening_trial);
 
-  // Compute the correction stress
+  // compute the correction stress
   double p_multiplier = 0.;
   Vector6d dp_dsigma_final = Vector6d::Zero();
   bool update_pds = false;
 
   // check if it is a loading or an unloading process
   double dyfun_t = yield_function_trial(0) - yield_function(0);
   double dyfun_s = yield_function_trial(1) - yield_function(1);
-
   // Yield is in tension or shear and dyfun_t and dyfun_s are greater than 0
   if ((yield_type == FailureState::Tensile && dyfun_t > 0) ||
       (yield_type == FailureState::Shear && dyfun_s > 0)) {
     p_multiplier = lambda;
     dp_dsigma_final = dp_dsigma;
-    update_pds = true;
+    // Plastic deviatoric strain is updated only in shear
+    if (yield_type == FailureState::Shear) update_pds = true;
   }
 
   // If yield type is elastic, but yield trial is not elastic
-  if (yield_type == FailureState::Elastic &&
-      yield_type_trial != FailureState::Elastic) {
+  else if (yield_type == FailureState::Elastic &&
```

Comment:
Could we just have it as an `if` with the appropriate conditions? or add an `else`?

Reply:
Yes, I think using if is the same